drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1529 @@
+	}
+
+#endif /* CONFIG_TMAG5273_TRIGGER */
+
+	retval = tmag5273_init_device_config(drv_cfg);
+
+	if (retval < 0) {
+		LOG_DBG("error setting device configuration |%d|", retval);
+		return retval;
+	}
+
+	return 0;
+}
+
+/* ----------------------------------------------------------- */
+
+static const struct sensor_driver_api tmag5273_driver_api = {
+	.attr_set = tmag5273_attr_set,
+	.attr_get = tmag5273_attr_get,
+#if CONFIG_TMAG5273_TRIGGER
+	.trigger_set = tmag5273_trigger_set,
+#endif
+	.sample_fetch = tmag5273_sample_fetch,
+	.channel_get = tmag5273_channel_get,
+};
+
+/* ----------------------------------------------------------- */
+
+#if CONFIG_TMAG5273_TRIGGER
+
+	/** Instantiation macro for trigger related options */
+	#define TMAG5273_CFG_TRIGGER(inst) \
+		.int_mode = DT_INST_PROP(inst, int_mode), \
+		.int_thresh_cnt = DT_INST_PROP(inst, int_threshold_count), \
+		.int_tresh_direction = DT_INST_PROP(inst, int_threshold_direction), \
+		.int_latched = DT_INST_PROP(inst, int_latched),
+
+#else
+
+	#define TMAG5273_CFG_TRIGGER(inst)
+
+#endif /* CONFIG_TMAG5273_TRIGGER */
+
+#if CONFIG_PM_DEVICE
+/** Instantiation macro for power management related options */
+	#define TMAG5273_CFG_PM(inst) \
+		.pm_suspend_mode = DT_INST_PROP(inst, pm_suspend_mode), \
+		.pm_sleeptime = DT_INST_PROP(inst, pm_sleeptime), \
+		.pm_i2c_workaround = DT_INST_PROP(inst, pm_i2c_vcc_slew_rate_sequence),
+
+#else
+
+	#define TMAG5273_CFG_PM(inst)
+
+#endif /* CONFIG_PM_DEVICE */
+
+/** Instantiation macro */
+#define TMAG5273_DEFINE(inst) \
+	static struct tmag5273_config tmag5273_driver_cfg##inst = { \
+		.i2c = I2C_DT_SPEC_INST_GET(inst), \
+		.x_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_X), \
+		.y_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_Y), \
+		.z_axis = (bool)(DT_INST_PROP(inst, axis) & TMAG5273_AXIS_Z), \
+		.temperature = DT_INST_PROP(inst, temperature), \
+		.meas_range = DT_INST_PROP(inst, range), \
+		.temperature_coefficient = DT_INST_PROP(inst, temperature_coefficient), \
+		.angle_magnitude_axis = DT_INST_PROP(inst, angle_magnitude_axis), \
+		.ch_mag_gain_correction = DT_INST_PROP(inst, ch_mag_gain_correction), \
+		.operation_mode = DT_INST_PROP(inst, operation_mode), \
+		.averaging = DT_INST_PROP(inst, average_mode), \
+		.trigger_conv_via_int = DT_INST_PROP(inst, trigger_conversion_via_int), \
+		.crc_enabled = DT_INST_PROP(inst, crc_enabled), \
+		.low_power_mode = DT_INST_PROP(inst, low_power), \
+		.mask_intb_value = \
+			(DT_INST_PROP(inst, trigger_conversion_via_int) \
+				&& !DT_INST_PROP(inst, force_mask_int_pin) ? \
+			TMAG5273_INT_MASK_INTB_ENABLED : TMAG5273_INT_MASK_INTB_MASKED), \
+		.need_int_pin = \
+			(DT_INST_PROP(inst, trigger_conversion_via_int) \
+			|| IS_ENABLED(CONFIG_TMAG5273_TRIGGER)), \
+		TMAG5273_CFG_TRIGGER(inst) \
+		TMAG5273_CFG_PM(inst) \
+	}; \
+	static struct tmag5273_data tmag5273_driver_data##inst = { \
+		.int_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int_gpios, { 0 }), \
```

Comment:
It seems like this would be constant as well. If that's the case, can you please add it to `struct tmag5273_config`?